ui/src/components/Drawer/index.vue

```diff
@@ -1,136 +1,221 @@
+
+const handleReconnect = () => {
+  if (timeoutId.value) {
+    clearTimeout(timeoutId.value);
+    timeoutId.value = null;
+  }
+
+  hasToken.value = false;
+  showEmpty.value = false;
+  fileManagerToken.value = '';
+  isRequestingToken.value = true;
+
+  lunaCommunicator.sendLuna(LUNA_MESSAGE_TYPE.CREATE_FILE_CONNECT_TOKEN, '');
+
+  timeoutId.value = setTimeout(() => {
+    if (!hasToken.value && isRequestingToken.value) {
+      showEmpty.value = true;
+      isRequestingToken.value = false;
+    }
+  }, MAX_WAIT_TIME);
+};
+
+onMounted(() => {
+  lunaCommunicator.onLuna(LUNA_MESSAGE_TYPE.OPEN, handleOpenDrawer);
+  lunaCommunicator.onLuna(LUNA_MESSAGE_TYPE.GET_FILE_CONNECT_TOKEN, handleCreateFileConnectToken);
+  lunaCommunicator.onLuna(LUNA_MESSAGE_TYPE.PING, () => {
+    const disbaleFileManager = lunaCommunicator.getDisbaleFileManager();
+    isDisableFileManager.value = disbaleFileManager;
+  });
+
+  const initialDisbaleFileManager = lunaCommunicator.getDisbaleFileManager();
+  if (initialDisbaleFileManager) {
+    isDisableFileManager.value = initialDisbaleFileManager;
+  }
+
+  mittBus.on('open-setting', () => {
+    drawerStatus.value = !drawerStatus.value;
+  });
+});
+
+onUnmounted(() => {
+  if (timeoutId.value) {
+    clearTimeout(timeoutId.value);
+    timeoutId.value = null;
+  }
+
+  lunaCommunicator.offLuna(LUNA_MESSAGE_TYPE.OPEN, handleOpenDrawer);
+  lunaCommunicator.offLuna(LUNA_MESSAGE_TYPE.GET_FILE_CONNECT_TOKEN, handleCreateFileConnectToken);
+  lunaCommunicator.offLuna(LUNA_MESSAGE_TYPE.PING);
+});
 </script>
+
+<template>
+  <n-drawer
+    id="drawer-inner-target"
+    resizable
+    placement="right"
+    :show="true"
+    :show-mask="false"
+    :default-width="502"
+    :min-width="502"
+    :max-width="800"
+    class="relative"
+    :style="{
+      display: drawerStatus ? 'block' : 'none',
+      opacity: drawerStatus ? 1 : 0,
+      top: '1px',
+    }"
+  >
+    <n-drawer-content :native-scrollbar="false">
+      <template #header>
+        <n-flex align="center" justify="space-between">
+          <n-text depth="1">
+            {{ connectionStore.assetName }}
+          </n-text>
+          <X class="cursor-pointer" :size="18" @click="closeDrawer" />
+        </n-flex>
+      </template>
+
+      <n-tabs size="medium" type="line" :default-value="filteredDrawerTabs[0].name" @update:value="handleTabChange">
+        <n-tab-pane v-for="tab in filteredDrawerTabs" :key="tab.name" display-directive="show" :name="tab.name">
+          <template #tab>
+            <n-flex align="center">
+              <component :is="tab.icon" :size="16" />
+              <span>{{ tab.label }}</span>
+            </n-flex>
+          </template>
+
+          <component
+            :is="tab.component"
+            :sftp-token="fileManagerToken"
+            :show-empty="showEmpty"
+            @reconnect="handleReconnect"
+          />
+        </n-tab-pane>
+      </n-tabs>
+    </n-drawer-content>
+  </n-drawer>
+</template>
```

Comment:
This code seems to be valid but slightly modified from the original template. Here's an overview of what was done or changed:

- Used `defineProps` before importing the props object.
- Removed unused imports at the top (`import { reactive, ref, computed, watch } from 'vue';`). It should only expose these utilities within Vue components that need them.

If there are no issues found with this change, you may proceed by reviewing it for readability. If adjustments have been made, they could potentially improve understanding and maintainability. However, without additional context about what might need improvement, I would describe the changes above briefly as minimal modifications.